pkg/uploader/logger.go

```diff
@@ -0,0 +1,104 @@
+package uploader
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"unicode"
+)
+
+type Dclssue struct {
+	Line        int    `json:"line"`
+	Column      int    `json:"column"`
+	Message     string `json:"message"`
+	SyntaxError bool   `json:"syntaxError"`
+	Severity    string `json:"severity"`
+	Code        int    `json:"code"`
+}
+type CompileError struct {
+	Status       string   `json:"status"`
+	Message      string   `json:"message"`
+	Errors       []string `json:"errors"`
+	CompileError struct {
+		DclIssues map[string][]Dclssue `json:"dclIssues"`
+	} `json:"compile_error"`
+}
+
+const severityError = "ERROR"
+const severityWarning = "WARNING"
+
+func (up *uploader) logResponse(res *http.Response) {
+
+	if res.StatusCode == http.StatusOK || res.StatusCode == http.StatusCreated || res.StatusCode == http.StatusNotModified {
+		up.log.Info("Base DCLs uploaded and compiled successfully")
+		return
+	} else if res.StatusCode == http.StatusBadRequest {
+		var ce CompileError
+		err := json.NewDecoder(res.Body).Decode(&ce)
+		if err == nil {
+			err = up.printCompileError(ce)
+			if err == nil {
+				return
```

Comment:
If we didn't manage to log the error properly, let's log the response body without any additional info.

Reply:
I wonder how to handle this error. Because we don't want to show it to the consumer. We should write it into some logs only we could see anyhow